tools/smithy-rs-sync/Cargo.lock

```diff
@@ -0,0 +1,366 @@
+# This file is automatically @generated by Cargo.
```

Comment:
Is it expected to commit Cargo.lock in vcs?

Reply:
Yup https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

Reply:
Ah, TIL.

Reply:
In my opinion, yes because it's a binary crate. If it was a library I wouldn't have committed it. I'm cool removing it if y'all want.

Reply:
https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries